### DISCUSSION TITLE
unable to add worker node to a 4.8 cluster that was initially created with 4.5

### DISCUSSION THREAD

**Discussion by kai-uwe-rommel** [okd-project/okd#1034] [Q&A]
**Describe the bug**
We created a cluster with OKD 4.5 initially in October 2020 and later updated it as new versions were released. Currently the cluster is at 4.8-2021-10-24. Around September 1st when the cluster was at 4.7 we successfully added a new worker node to it. Now I tried to add another new worker node but its ignition fails with a certificate error:

![grafik](https://user-images.githubusercontent.com/26381804/146196570-805d4a86-72ab-4540-8cfd-4f6a05fb783d.png)

**Version**
4.8-2021-10-24 on vSphere UPI, the booting FCOS is 34.20210626

**How reproducible**
100%

**Log bundle**
Will provide if needed, but since the node does not boot, I don't see how I could.

**Comment by kai-uwe-rommel** [comment]
The new node VM gets a stub ignition file via guestinfo VM property and then merges the actual (small) worker.ign from the HTTP server. This then fails at this step where it merges the actual ignition data from the machine config server.

So it looks like technology advanced and somewhere it now insists on SAN attributes in a certificate where there are none.
What can I do to fix this?

**Comment by kai-uwe-rommel** [comment]
My workaround/solution now was to start the new nodes with the same old FCOS 32 that was current when the cluster was initially created. This worked and MCO then updated the new nodes to current FCOS 34 properly.

**Comment by kai-uwe-rommel** [comment]
So this can become a problem to anyone running such a cluster for more that a couple of (few) months and wants to scale it now.

**Comment by kai-uwe-rommel** [comment]
@vrutkovs, any comment?

**Comment by vrutkovs** [comment]
Please attach (or upload to the public file sharing service) [must-gather](https://docs.okd.io/latest/support/gathering-cluster-data.html#support_gathering_data_gathering-cluster-data) archive

**Comment by kai-uwe-rommel** [comment]
Here is one: https://domino.ars.de/file/must-gather.local.9150851182498598657.tgz
(Download available for 7 days.)

**Comment by kai-uwe-rommel** [comment]
(Download made available again for another 7 days.)

**Comment by rvanderp3** [comment]
I believe this was a [deprecation in golang](https://bugzilla.redhat.com/show_bug.cgi?id=1886892).  I think the best path forward would be to update the certificate to use a SAN rather than the common name.

**Comment by kai-uwe-rommel** [comment]
Sure I would also rather like to update the certificate ... if anyone points me to instructions on how to do it!
The certificate of the machine config server is a 10-year valid certificate generated internally by the cluster's "rootca"...
How to renew?

**Comment by rvanderp3** [comment]
ah .... yes, it is.  i'll look in to that.

**Comment by rvanderp3** [comment]
I took a look at one of my clusters and it looks like there are a couple of options I can think of:
1. If you still have the installation directory from the 4.5 installation(I realize this is unlikely) you can find the root-ca cert/key pair in `.openshift_install_state.json` and generate a new certificate. Once generated you can update the secret `machine-config-server-tls` in the `openshift-machine-config-operator` and restart the machine-config-server pods.
2. Generate a new certificate with a different CA and update the machine-config-server-tls secret as well as update the CA certificate `guestinfo.ignition.config.data` with the CA cert that signed the cert.

I couldn't find the root CA key on the cluster itself but I may have just missed it.

**Comment by kai-uwe-rommel** [comment]
I do indeed still have the installation directory.
Otherwise the root-ca is stored as the "root-ca" CM in the kube-system namespace.
But that contains only its certificate, not the key, as you wrote. Hmm.

**Reply by rvanderp3** [reply]
The certificate is, but not the key(atleast on my cluster).  You'll need the key to create a new cert.

**Comment by kai-uwe-rommel** [comment]
I thought it could be somewhere under /etc/kubernetes/static-pod-resources but nothing there really catched my eye.

**Comment by rvanderp3** [comment]
Yep, I checked there as well and a few other places on-disk.  It looks like it's only used to sign certificates pre-bootstrap.

**Comment by kai-uwe-rommel** [comment]
Since the OKD team is not going to fix this and there is the relatively easy workaround of keeping the FCOS template with which the cluster was originally built, it now makes sense to close this.
